### PR TITLE
Create-note

### PR DESCRIPTION
**Pull request by prajapati-priyanka**
Add Note functionality
 - User can create notes by choosing given label , priority and colours.
 - User can pin and unpin notes. 
 - when note is pinned, it also reflects into the pinned section, but when unpinned it will not show in the pinned section.

Things To Work Upon:
 - Responsiveness. It will be handled in the end after completing all the functionalities.

### CODE REVIEW

File: src/reducer/notes-reducer.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const notesReducer = (state, action)=>{
+    switch (action.type) {
+        case "ADD_NOTE":
+            return{
+                ...state,
+                notes: action.payload
+            }
+
```
Comment (vermasavita) [inline src/reducer/notes-reducer.js:8]: remove these spaces

File: src/components/createNote/CreateNote.jsx [hunk 2]
```diff
@@ -1,54 +1,106 @@
...
+  const [noteContent, setNoteContent] = useState({
+    title: "",
+    content: "",
+    label: "",
+    priority: "",
+    noteColor: "",
+  });
+
+  
+
```
Comment (vermasavita) [inline src/components/createNote/CreateNote.jsx]: Remove the space

File: src/components/createNote/CreateNote.jsx [hunk 3]
```diff
@@ -1,54 +1,106 @@
...
+       setNoteContent((prevData) => {
+      return {
+        ...prevData,
+        [name]: value,
+      };
+    });
+  };
+
+
+
```
Comment (vermasavita) [inline src/components/createNote/CreateNote.jsx:25]: Remove the space

File: src/pages/home/Home.jsx [hunk 4]
```diff
@@ -1,8 +1,14 @@
 import { FaSearch } from "react-icons/fa";
 import { SideNav, NoteCard, CreateNote,Filter } from "../../components";
 import "./Home.css";
+import { useNote } from "../../context/notes-context";
 
 const Home = () => {
+  
+  const {notesState} = useNote();
+  // notesState.notes = JSON.parse(localStorage.getItem("notes"))
```
Comment (vermasavita) [inline src/pages/home/Home.jsx]: remove commented space

**Review by vermasavita — APPROVED** [review]
Functionalities working fine.

File: src/components/createNote/CreateNote.jsx [hunk 5]
```diff
@@ -1,54 +1,128 @@
...
+  };
+
+  const [noteContent, setNoteContent] = useState(initialNotesData);
+
+  const expandColorPallete = () => {
+    setIsOpen((isOpen) => !isOpen);
+  };
+
+  return (
+    <>
```
Comment (palash23jain) [inline src/components/createNote/CreateNote.jsx:27]: good use of fragment

**Review by palash23jain — APPROVED** [review]
Code looks neat and clean. Good to go

**Pull request merged**